File: replace_colors.js
const fs = require('fs');
const path = require('path');

const replacements = [
    { pattern: /background:\s*["']white["']/g, repl: 'background: "var(--bg-card)"' },
    { pattern: /background:\s*["']#ffffff["']/g, repl: 'background: "var(--bg-primary)"' },
    { pattern: /background:\s*["']#F9FAFB["']/g, repl: 'background: "var(--bg-secondary)"' },
    { pattern: /background:\s*["']#F3F4F6["']/g, repl: 'background: "var(--bg-tertiary)"' },
    { pattern: /color:\s*["']#111827["']/g, repl: 'color: "var(--text-primary)"' },
    { pattern: /color:\s*["']#4B5563["']/g, repl: 'color: "var(--text-secondary)"' },
    { pattern: /color:\s*["']#6B7280["']/g, repl: 'color: "var(--text-tertiary)"' },
    { pattern: /border:\s*["']1px solid #E5E7EB["']/g, repl: 'border: "1px solid var(--border-primary)"' },
    { pattern: /borderBottom:\s*["']1px solid #E5E7EB["']/g, repl: 'borderBottom: "1px solid var(--border-primary)"' },
    { pattern: /borderTop:\s*["']1px solid #E5E7EB["']/g, repl: 'borderTop: "1px solid var(--border-primary)"' },
    { pattern: /borderRight:\s*["']1px solid #E5E7EB["']/g, repl: 'borderRight: "1px solid var(--border-primary)"' },
    { pattern: /borderColor:\s*["']#E5E7EB["']/g, repl: 'borderColor: "var(--border-primary)"' },
];

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else {
            if (file.endsWith('.tsx')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walk('./src');

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let newContent = content;
    
    replacements.forEach(({pattern, repl}) => {
        newContent = newContent.replace(pattern, repl);
    });
    
    if (newContent !== content) {
        fs.writeFileSync(file, newContent, 'utf8');
        console.log('Updated', file);
    }
});

File: replace_colors.py
import os
import glob
import re

replacements = [
    (r'background:\s*[\"\'\']white[\"\'\']', 'background: "var(--bg-card)"'),
    (r'background:\s*[\"\'\']#ffffff[\"\'\']', 'background: "var(--bg-primary)"'),
    (r'background:\s*[\"\'\']#F9FAFB[\"\'\']', 'background: "var(--bg-secondary)"'),
    (r'background:\s*[\"\'\']#F3F4F6[\"\'\']', 'background: "var(--bg-tertiary)"'),
    (r'color:\s*[\"\'\']#111827[\"\'\']', 'color: "var(--text-primary)"'),
    (r'color:\s*[\"\'\']#4B5563[\"\'\']', 'color: "var(--text-secondary)"'),
    (r'color:\s*[\"\'\']#6B7280[\"\'\']', 'color: "var(--text-tertiary)"'),
    (r'border:\s*[\"\'\']1px solid #E5E7EB[\"\'\']', 'border: "1px solid var(--border-primary)"'),
    (r'borderBottom:\s*[\"\'\']1px solid #E5E7EB[\"\'\']', 'borderBottom: "1px solid var(--border-primary)"'),
    (r'borderTop:\s*[\"\'\']1px solid #E5E7EB[\"\'\']', 'borderTop: "1px solid var(--border-primary)"'),
    (r'borderRight:\s*[\"\'\']1px solid #E5E7EB[\"\'\']', 'borderRight: "1px solid var(--border-primary)"'),
    (r'borderColor:\s*[\"\'\']#E5E7EB[\"\'\']', 'borderColor: "var(--border-primary)"'),
]

files = glob.glob('src/**/*.tsx', recursive=True)

for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    new_content = content
    for pattern, repl in replacements:
        new_content = re.sub(pattern, repl, new_content)
    
    if new_content != content:
        with open(file, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f'Updated {file}')
